Add Ratings component and Reviews link to Theme screen

epeicher opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #https://github.com/Automattic/dotcom-forge/issues/4561

## Proposed Changes

* Extract the `ReviewsSummary` to its own component
* Add the Rating component including the average rating
* Add the "X Reviews" link

|Before | After|
|-------|------|
|      ![CleanShot 2023-12-01 at 18 17 01@2x](https://github.com/Automattic/wp-calypso/assets/3519124/6c088247-bb3f-4445-8272-7c46ab9e4632)        |      ![CleanShot 2023-12-01 at 18 16 48@2x](https://github.com/Automattic/wp-calypso/assets/3519124/abecaff8-5997-472d-aa42-f19b28b22f6d)      |

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* With the following flags enabled `marketplace-reviews-show` and `marketplace-add-review`
* Navigate to a Theme details page with reviews, for example `/themes/hevor`
* Make sure the Rating component, Reviews link and Add Review button are displayed

- Navigate to a Theme without reviews, for example `/themes/makoney`
- Make sure you do not see the Rating and Reviews link components

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

epeicher committed:
Extract ReviewsSummary to its own component

epeicher committed:
Add Rating component and reviews link to Themes

epeicher committed:
Merge branch 'trunk' into add/theme-reviews-summary-with-button

epeicher committed:
Add padding for smaller screens

WBerredo reviewed (changes requested):
There seems to be a regression on the layout for themes without reviews. Now its not aligned with the other items.

![CleanShot 2023-12-05 at 09 48 49@2x](https://github.com/Automattic/wp-calypso/assets/5039531/8ec4a22e-a95a-416b-8153-73cecb04ed29)

WBerredo commented on client/my-sites/marketplace/components/reviews-summary/index.tsx:
```diff
@@ -0,0 +1,73 @@
+						</Button>
+					) }
+				</div>
+				<Button onClick={ () => setIsVisible( true ) }>{ translate( 'Add Review' ) }</Button>
```
Should we already use the `canPublishThemeReview` selector to define if a user can add a review or not?

epeicher replied:
That was going to be tackled as part of 
- https://github.com/Automattic/dotcom-forge/issues/4737

but I might use it here to avoid the extra padding reported above.

WBerredo replied:
But we would still need to check the extra padding issue in the scenario of:
* The product doesn't have any reviews
* The user can publish a review

WBerredo replied:
> That was going to be tackled as part of

Great, that's fine if its already being tracked

epeicher commented:
> There seems to be a regression on the layout for themes without reviews. Now its not aligned with the other items.

That's right, some additional padding has been added for the reviews component in small screens, I'll address that 👍

epeicher committed:
Remove empty div when no reviews available

This fixes the extra gap appearing on the left of the button

epeicher commented:
@WBerredo, I have removed the extra `gap` in https://github.com/Automattic/wp-calypso/pull/84775/commits/04d8ae65305ea0903c7fa8db40086eca1b2e43ec and I have added your suggestion to https://github.com/Automattic/dotcom-forge/issues/4737#issuecomment-1841008086 so it can be used if required. Thanks for the suggestion!

WBerredo reviewed (approved):
Thanks for applying the suggestions, all LGTM now! 🚢

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9901201).
